main.tf

```diff
@@ -38,7 +38,7 @@ resource "google_service_account" "service_accounts" {
   for_each     = local.names
   account_id   = "${local.prefix}${lower(each.value)}"
   display_name = var.display_name
-  description  = var.description
+  description  = index(local.names, each.value) > length(var.descriptions) - 1 ? var.description : element(var.descriptions, index(local.names, each.value))
```

Comment:
This raises an error:

```
         on ../../../main.tf line 41, in resource "google_service_account" "service_accounts":
         41:   description  = index(local.names, each.value) > length(var.descriptions) - 1 ? var.description : element(var.descriptions, index(local.names, each.value))
           |----------------
           | each.value is "test-first"
           | local.names is set of string with 2 elements
       
       Call to function "index" failed: argument must be a list or tuple.
```

Reply:
Ah. Forgot that index doesn't like sets. And now that brings up ordering consistency with the set/list type conversion. I think I'll swap to use var.names instead of local.names so there's no invisible force moving the elements around.

Reply:
Yeah, I'm actually concerned we might have issues with consistent ordering. Perhaps `descriptions` should be a map of name->description instead?

Reply:
I had originally thought about using a map but didn't like the way it looked once I actually set it up. Plus maps seem to be used somewhat rarely in the Google modules (strings with lambda operator get used somewhat regularly though - which does make integration with the modules like the IAM bindings module somewhat tedious).

If you want I can change it to a map of `{sa_name:description}` pairs. Otherwise using `var.names` should prevent ordering issues (at the expense of only picking the 'first' description if an SA name is defined twice in the `var.names` list).

Reply:
`var.names` will be consistent, but the `index` of each is potentially inconsistent because `set` doesn't provide ordering guarantees.

Let's go ahead and switch to a map, thanks!

Reply:
`var.names` is a list though so it _should_ be consistent. `local.names` is the set version of the list that would provoke indexing problems. I won't introduce errors by changing the for_each to use the list, but as long as the value lookups are done against the list it should be consistent.
I can still setup the map if you want though. :D

Reply:
Good point actually, this should be fine.